datafusion/expr/src/expr.rs

```diff
@@ -385,6 +408,20 @@ impl Cast {
     }
 }
 
+/// Cast expression
```

Comment:
Can we please add documentation about what `PromotePrecision` does and in what circumstances it is needed?